src/platforms/gbm-kms/server/kms/quirks.cpp

```diff
@@ -58,52 +58,37 @@ class mgg::Quirks::Impl
         {
-            // Quirk format is skip:(devnode|driver):value
+            auto const disable_kms_probe = "disable-kms-probe:";
 
-            if (quirk.size() < 5)
+            if (quirk.starts_with("skip:"))
             {
-                mir::log_warning(
-                    "Ignoring unexpected value for %s option: %s (expects value of the form “skip:<type>:<value>”)",
-                    quirks_option_name,
-                    quirk.c_str());
-                continue;
+                // Quirk format is skip:(devnode|driver):value
+                auto type_delimeter_pos = quirk.find(':', 5);
+                auto const type = quirk.substr(5, type_delimeter_pos - 5);
+                auto const value = quirk.substr(type_delimeter_pos + 1);
+                if (type == "devnode")
+                {
+                    devnodes_to_skip.insert(value);
+                    continue;
+                }
+                else if (type == "driver")
+                {
+                    drivers_to_skip.insert(value);
+                    continue;
+                }
             }
-            if (strncmp(quirk.c_str(), "skip:", strlen("skip:")) != 0)
+            else if (quirk.starts_with(disable_kms_probe))
             {
-                mir::log_warning(
-                    "Ignoring unexpected value for %s option: %s (expects value of the form “skip:<type>:<value>”)",
-                    quirks_option_name,
-                    quirk.c_str());
+                // Quirk format is disable-kms-probe:value
+                skip_modesetting_support.emplace(quirk.substr(strlen(disable_kms_probe)));
                 continue;
             }
 
-            auto type_delimeter_pos = quirk.find(':', 5);
-            if (type_delimeter_pos == std::string::npos)
-            {
-                mir::log_warning(
-                    "Ignoring unexpected value for %s option: %s (expects value of the form “skip:<type>:<value>”)",
-                    quirks_option_name,
-                    quirk.c_str());
-                continue;
-            }
-            auto const type = quirk.substr(5, type_delimeter_pos - 5);
-            auto const value = quirk.substr(type_delimeter_pos + 1);
-            if (type == "devnode")
-            {
-                devnodes_to_skip.insert(value);
-            }
-            else if (type == "driver")
-            {
-                drivers_to_skip.insert(value);
-            }
-            else
-            {
-                mir::log_warning(
-                    "Ignoring unexpected value for %s option: %s (valid types are “devnode” and “driver”)",
-                    quirks_option_name,
-                    type.c_str());
-                continue;
-            }
+            // If we didn't `continue` above, we're ignoring...
+            mir::log_warning(
+                "Ignoring unexpected value for %s option: %s (expects value of the form “skip:<type>:<value>”)",
```

Comment:
This is no longer true; it expects either “skip:<type>:<value>” or “disable-kms-probe:<value>”.

Next type of quirk we add we should make a more principled “here's what the supported quirks are, and how to specify them” infrastrucutre :wink: